presto-main/src/test/java/io/prestosql/execution/TestGrantTask.java

```diff
@@ -0,0 +1,108 @@
+
+import java.util.Optional;
+import java.util.Set;
+
+import static io.prestosql.connector.CatalogName.createInformationSchemaCatalogName;
+import static io.prestosql.connector.CatalogName.createSystemTablesCatalogName;
+import static io.prestosql.metadata.MetadataManager.createTestMetadataManager;
+import static io.prestosql.sql.tree.PrincipalSpecification.Type.ROLE;
+import static io.prestosql.testing.TestingSession.testSessionBuilder;
+import static io.prestosql.transaction.InMemoryTransactionManager.createTestTransactionManager;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class TestGrantTask
+{
+    @Test
+    public void testGrantToNonExistingRole()
+    {
+        CatalogManager catalogManager = new CatalogManager();
+        catalogManager.registerCatalog(createBogusTestingCatalog("test"));
+        TransactionManager transactionManager = createTestTransactionManager(catalogManager);
+        Session testSession = testSessionBuilder()
+                .setTransactionId(transactionManager.beginTransaction(false))
+                .setCatalog("test")
+                .build();
+        MetadataManager metadata = createTestMetadataManager(transactionManager, new FeaturesConfig());
+        AccessControl accessControl = new AllowAllAccessControl();
+
+        Grant statement = new Grant(Optional.empty(), Optional.empty(), QualifiedName.of("test.schema.ignore"), new PrincipalSpecification(ROLE, new Identifier("non-existent")), false);
+
+        assertThatThrownBy(() -> new GrantTask().execute(statement, transactionManager, metadata, accessControl, testSession, ImmutableList.of()))
+                .isInstanceOf(PrestoException.class)
+                .hasMessageContaining("Role 'non-existent' does not exist");
+    }
+
+    public static Catalog createBogusTestingCatalog(String catalogName)
+    {
+        CatalogName catalog = new CatalogName(catalogName);
+        return new Catalog(
+                catalogName,
+                catalog,
+                createTestSessionConnector(),
+                createInformationSchemaCatalogName(catalog),
+                createTestSessionConnector(),
+                createSystemTablesCatalogName(catalog),
+                createTestSessionConnector());
+    }
+
+    private static Connector createTestSessionConnector()
+    {
+        return new Connector()
+        {
+            @Override
+            public ConnectorTransactionHandle beginTransaction(IsolationLevel isolationLevel, boolean readOnly)
+            {
+                return new ConnectorTransactionHandle() {};
+            }
+
+            @Override
+            public ConnectorMetadata getMetadata(ConnectorTransactionHandle transaction)
+            {
+                return new SystemTablesMetadataMock(new StaticSystemTablesProvider(ImmutableSet.of()));
+            }
+        };
+    }
+
+    private static class SystemTablesMetadataMock
+            extends
+            SystemTablesMetadata
```

Comment:
Formatting is odd. Looks like someone is using new computer.

Reply:
Nope, this is a human error. My formatting options are fine.